py4DSTEM/process/diffraction/crystal_calibrate.py

```diff
@@ -0,0 +1,465 @@
+            self.coef_update = coef_update
+
+        def get_coefs(
+            self,
+            coefs_fit,
+        ):
+            coefs = np.zeros_like(coefs_fit)
+            for a0 in range(6):
+                if self.coef_update[a0]:
+                    coefs[a0] = coefs_fit[self.coef_index[a0]]
+                else:
+                    coefs[a0] = self.coefs_init[a0]
+            coefs[6:] = coefs_fit[6:]
+            
+            return coefs
+            
+        def fitfun(self, k, *coefs_fit):
+            coefs = self.get_coefs(coefs_fit=coefs_fit)
+
+            # Update g vector positions
+            a, b, c = coefs[:3]
+            alpha = np.deg2rad(coefs[3])
+            beta = np.deg2rad(coefs[4])
+            gamma = np.deg2rad(coefs[5])
+            f = np.cos(beta) * np.cos(gamma) - np.cos(alpha)
+            vol = a*b*c*np.sqrt(1 \
+                + 2*np.cos(alpha)*np.cos(beta)*np.cos(gamma) \
+                - np.cos(alpha)**2 - np.cos(beta)**2 - np.cos(gamma)**2)
+            lat_real = np.array(
+                [
+                    [a,               0,                 0],
+                    [b*np.cos(gamma), b*np.sin(gamma),   0],
+                    [c*np.cos(beta), -c*f/np.sin(gamma), vol/(a*b*np.sin(gamma))],
+                ]
+            )
+            # Inverse lattice, metric tensors
+            metric_real = lat_real @ lat_real.T
+            metric_inv = np.linalg.inv(metric_real)
+            lat_inv = metric_inv @ lat_real
+            g_vec_all = (self.hkl.T @ lat_inv).T
+            g_vec_leng = np.linalg.norm(g_vec_all, axis=0)        
+        
+            # Calculate fitted intensity profile
+            k_broadening = coefs[6]
+            int_scale = coefs[7:]
+            int_sf = calc_1D_profile(
+                k,
+                g_vec_leng,
+                self.struct_factors_int,
+                k_broadening=k_broadening,
+                int_scale=int_scale,
+                normalize_intensity=False,
+            )     
+
+            return int_sf
+
+
+    if fit_all_intensities:
+        fit_crystal = FitCrystal(
+            self,
+            coef_index = coef_index,
+            coef_update = coef_update,
+            fit_all_intensities=fit_all_intensities,
+        )
+        
+        coefs = (
+            *tuple(self.cell),
+            k_broadening,
+            *tuple(np.ones(self.g_vec_leng.shape[0])),
+        )
+        bounds = (0.0, np.inf)
+        popt, pcov = curve_fit(
+            fit_crystal.fitfun, 
+            k, 
+            int_exp, 
+            p0 = coefs,
+            bounds = bounds,
+            )
+    else:
+        coefs = (
+            *tuple(self.cell),
+            k_broadening,
+            1.0,
+        )
+        bounds = (0.0, np.inf)
+        popt, pcov = curve_fit(
+            fit_crystal.fitfun, 
+            k, 
+            int_exp, 
+            p0 = coefs,
+            bounds = bounds,
+            )
```

Comment:
It's possible i've just missed something, but: it looks to me like `fit_crystal` is never defined if we enter into this `else:` conditional.  Is that right?  If so this should be fixed